### PR TITLE
Make threading an explicit option rather than relying on the MPI impl…

### PR DESCRIPTION
**Pull request by ocaisa**
…ementation (ParaStaion MPI can support both possibilities)

### CODE REVIEW

**Comment by ocaisa** [comment]
Resolves #795

File: easybuild/easyblocks/s/scotch.py [hunk 1]
```diff
@@ -108,7 +117,7 @@ def build_step(self):
         if self.toolchain.options['i8']:
             cflags += " -DINTSIZE64"
 
-        if not self.toolchain.mpi_family() in [toolchain.INTELMPI, toolchain.QLOGICMPI]:  #@UndefinedVariable
+        if self.cfg['threadedmpi']:
```
Comment (boegel) [inline easybuild/easyblocks/s/scotch.py]: @ocaisa this implies that for builds on top of OpenMPI (for example), `threadedmpi = True` must be now specified to obtain the same installation, so this should be something like:

```python
if self.cfg['threadedmpi'] or if self.toolchain.mpi_family() not in [toolchain.INTELMPI, toolchain.QLOGICMPI]
```

to ensure backward compatibility?

Comment (ocaisa) [inline easybuild/easyblocks/s/scotch.py]: Ok, but I'm pretty sure this wasn't working for people, to my eyes it's mostly there for OpenFOAM and threaded scotch does not work with OpenFOAM.
I'll set the default to true for OpenMPI but personally I think this is probably not doing what people actually expect.

Comment (ocaisa) [inline easybuild/easyblocks/s/scotch.py]: How do I check a variable has been set in the easyconfig? I need to know if they have explicitly set the value to false in the easyconfig so I can ensure I only change the value to true when it's not defined there.

Comment (boegel) [inline easybuild/easyblocks/s/scotch.py]: @ocaisa you could use `None` as default value, and assume people will set it to `False` in the easyconfig?

and then check `self.cfg['threadedmpi'] is None`

there's no easy way to distinguish between a default value and the same value but specified by the easyconfig

**Comment by ocaisa** [comment]
@boegel Tested with ParaStation MPI and Intel MPI, backwards compatibility is now there and the new logic works

File: easybuild/easyblocks/s/scotch.py [hunk 2]
```diff
@@ -108,9 +117,12 @@ def build_step(self):
         if self.toolchain.options['i8']:
             cflags += " -DINTSIZE64"
 
-        if not self.toolchain.mpi_family() in [toolchain.INTELMPI, toolchain.QLOGICMPI]:  #@UndefinedVariable
+        if self.cfg['threadedmpi']: 
             cflags += " -DSCOTCH_PTHREAD"
-
+        # TODO For backwards compatability of v2.8.0 the following is necessary but could be removed on a major version upgrade
+        if self.cfg['threadedmpi'] is None and not self.toolchain.mpi_family() in [toolchain.INTELMPI, toolchain.QLOGICMPI]:
```
Comment (boegel) [inline easybuild/easyblocks/s/scotch.py]: please change to `not in`:

```python
and self.toolchain.mpi_family() not in [toolchain.INTELMPI, toolchain.QLOGICMPI]:
```

**Comment by boegel** [comment]
going in, thanks @ocaisa!

**Pull request merged**